tests/unit/ops/adagrad/test_cpu_adagrad.py

```diff
@@ -20,117 +21,125 @@ def check_equal(first, second, atol=1e-2, verbose=False):
     np.testing.assert_allclose(x, y, err_msg="param-update mismatch!", atol=atol)
 
 
-@pytest.mark.parametrize('model_size',
-                         [
-                             (64),
-                             (22),
-                             (55),
-                             (127),
-                             (1024),
-                             (1048576),
-                             (30000000),
-                         ]) # yapf: disable
-def test_cpu_adagrad_opt(model_size):
-    device = 'cpu'
-    rng_state = torch.get_rng_state()
-    param = torch.nn.Parameter(torch.randn(model_size, device=device))
-    torch.set_rng_state(rng_state)
-    param1 = torch.nn.Parameter(torch.randn(model_size, device=device))
-    torch.set_rng_state(rng_state)
-
-    optimizer = DeepSpeedCPUAdagrad([param])
-    optimizer1 = torch.optim.Adagrad([param1])
-
-    for i in range(10):
+class TestCPUAdagrad(DistributedTest):
+    world_size = 1
+    requires_cuda_env = False
+    if not torch.cuda.is_available():
+        init_distributed = False
+        set_dist_env = False
```

Comment:
I see this repeated in multiple tests. I think we could move it to `DistributedExec` or `DistributedTest` classes to avoid code duplication.